buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -2101,13 +2101,13 @@ class BeamModulePlugin implements Plugin<Project> {
         config.cleanupJobServer.mustRunAfter javaTask
 
         // Task for running testcases in Python SDK
-        def testOpts = [
-          "--attr=${config.pythonTestAttr}"
-        ]
         def beamPythonTestPipelineOptions = [
           "pipeline_opts": config.pythonPipelineOptions + sdkLocationOpt,
-          "test_opts": testOpts + config.nosetestsOptions,
-          "suite": "xlangValidateRunner"
+          "test_opts": config.pytestOptions,
+          "suite": "xlangValidateRunner",
+          "pytest": true, // TODO(BEAM-3713): Remove this once nose is removed.
+          // TODO(BEAM-10507): Remove no_xlang_transforms after fixing
```

Comment:
I think you can remove this TODO. We will always want to exclude tests marked with `no_xlang_transforms`.